app/actions/websocket/posts.ts

```diff
@@ -0,0 +1,204 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {Model} from '@nozbe/watermelondb';
+import {DeviceEventEmitter} from 'react-native';
+
+import {storeMyChannelsForTeam, markChannelAsUnread, markChannelAsViewed, updateLastPostAt} from '@actions/local/channel';
+import {markPostAsDeleted} from '@actions/local/post';
+import {fetchMyChannel, markChannelAsRead} from '@actions/remote/channel';
+import {fetchPostAuthors, fetchPostById} from '@actions/remote/post';
+import {ActionType, Events} from '@constants';
+import DatabaseManager from '@database/manager';
+import {queryMyChannel} from '@queries/servers/channel';
+import {queryPostById} from '@queries/servers/post';
+import {queryCurrentChannelId, queryCurrentUserId} from '@queries/servers/system';
+import {isFromWebhook, isSystemMessage, shouldIgnorePost} from '@utils/post';
+
+import type {WebSocketMessage} from '@typings/api/websocket';
+
+export async function handleNewPostEvent(serverUrl: string, msg: WebSocketMessage) {
+    const database = DatabaseManager.serverDatabases[serverUrl];
+    if (!database) {
+        return;
+    }
+
+    const post: Post = JSON.parse(msg.data.post);
```

Comment:
here we are relying on the fact that `msg.data.post` is defined and it is in fact a json object, reality is that JSON.parse can throw, so it would be probably safer to wrap this in a try/catch